service/datastore/internal/src/main/java/org/eclipse/kapua/service/datastore/internal/MessageStoreFacadeImpl.java

```diff
@@ -498,9 +498,6 @@ private boolean isAnyClientId(String clientId) {
     @Override
     public void refreshAllIndexes() throws ClientException {
         messageRepository.refreshAllIndexes();
-        clientInfoRepository.refreshAllIndexes();
-        channelInfoRepository.refreshAllIndexes();
-        metricInfoRepository.refreshAllIndexes();
     }
```

Comment:
Why removing the other three refreshes? The facade feels like the correct place to call refresh on all involved repositories at once

Reply:
Ah, I see now that the elasticsearch implementation of refreshAllIndexes really works on all indexes. Is this ok? Does this work with other implementations like opensearch (sounds strange to me that the "delete all" won't work, but "refresh all" is ok.
Conceptually, I'd look into refreshing only the indexes concerning the specific repository, as per the delete, and keep calling all four repositories from the facade.

Reply:
I think OpenSearch complains only about the deleteAllIndexes permissions and not refreshAll because of the possible danger of the first operation wrt the other (despite being admin logged)
I did not perform a similar operation as you propose because in our low-level client, there is not a method to refresh a precise index so I left just 1 call to the call to the "refresh all indexes"